ui/components/Navigator.js

```diff
@@ -703,70 +708,73 @@ class Navigator extends React.Component {
+      prevState.forEach((cat, ind) => {
+        if (cat.id === LIFECYCLE) {
+          cat.children.forEach((catc, ind1) => {
+            if (catc.id == SERVICE_MESH) {
+              return;
+            }
+            const icon = pickIcon(catc.id, catc.href);
+            prevState[ind].children[ind1].icon = icon;
 
-      if (cat.id === 'Configuration') {
-        let show = false;
-        cat.children?.forEach((ch) => {
-          if (ch.id === 'Designs') {
-            const idx = self.props.capabilitiesRegistry?.capabilities?.findIndex(
-              (cap) => cap.feature === 'persist-meshery-patterns',
-            );
-            if (idx != -1) {
-              ch.show = true;
-              show = true;
+            const cr = fetchChildren(catc.id);
+            prevState[ind].children[ind1].children = cr;
+          });
+        }
+
+        if (cat.id === 'Configuration') {
+          let show = false;
+          cat.children?.forEach((ch) => {
+            if (ch.id === 'Designs') {
+              const idx = props.capabilitiesRegistry?.capabilities?.findIndex(
+                (cap) => cap.feature === 'persist-meshery-patterns',
+              );
+              if (idx != -1) {
+                ch.show = true;
+                show = true;
+              }
             }
-          }
-        });
+          });
 
-        cat.show = show;
-      }
+          cat.show = show;
+        }
+      });
+      return prevState;
     });
-  }
-  updateAdaptersLink() {
-    const { navigatorComponents } = this.state;
-    navigatorComponents.forEach((cat, ind) => {
-      if (cat.id === LIFECYCLE) {
-        cat.children.forEach((catc, ind1) => {
-          if (
-            typeof navigatorComponents[ind].children[ind1].children[0] !== 'undefined' &&
-            typeof navigatorComponents[ind].children[ind1].children[0].href !== 'undefined'
-          ) {
-            const val = true;
-            const newhref = `${navigatorComponents[ind].children[ind1].children[0].href}`;
-            navigatorComponents[ind].children[ind1].link = val;
-            navigatorComponents[ind].children[ind1].href = newhref;
-          }
-        });
-      }
+  };
+
+  const updateAdaptersLink = () => {
+    setNavigatorComponents((prevState) => {
+      prevState.forEach((cat, ind) => {
+        if (cat.id === LIFECYCLE) {
+          cat.children.forEach((catc, ind1) => {
+            if (
+              typeof prevState[ind].children[ind1].children[0] !== 'undefined' &&
+              typeof prevState[ind].children[ind1].children[0].href !== 'undefined'
+            ) {
+              const val = true;
+              const newhref = `${prevState[ind].children[ind1].children[0].href}`;
+              prevState[ind].children[ind1].link = val;
+              prevState[ind].children[ind1].href = newhref;
+            }
+          });
+        }
+        return prevState;
+      });
     });
-  }
+  };
 
-  static getDerivedStateFromProps(props, state) {
```

Comment:
this is static function have you made sure we also update code if we are using this function outside of this file

Reply:
Can you specify what kind of changes? because I think being a lifecycle method, it should not be affected by its uses in other components.

Reply:
Yeah, however in Meshery we have use static functions outside of file as ref, I know not a good practice, however it would be nice to make sure that we haven't used this function other then this file

Reply:
getDerivedStateFromProps is used in other components. I will check its effects.